ports/esp32/machine_i2s.c

```diff
@@ -151,7 +151,7 @@ static uint32_t get_dma_buf_count(uint8_t mode, i2s_data_bit_width_t bits, forma
 
     uint32_t dma_buf_count = ibuf / (DMA_BUF_LEN_IN_I2S_FRAMES * dma_frame_size_in_bytes);
 
-    return dma_buf_count;
+    return MAX(dma_buf_count, 2);
```

Comment:
I think this requires a small comment like "ESP-IDF driver always requires at least two DMA buffers".